### DISCUSSION TITLE
failed to add another master to my raspberry pi cluaster

### DISCUSSION THREAD

**Discussion by MosheAzraf** [k3s-io/k3s#10072] [Q&A]
hi,
im trying to add another master to my cluster that runs on raspberry pi os lite, but i get error:
"Job for k3s.service failed because the control process exited with error code.
See "systemctl status k3s.service" and "journalctl -xeu k3s.service" for details."

the script im trying to follow is:
```curl -sfL https://get.k3s.io | K3S_URL=https://192.168.33.14:6443 K3S_TOKEN=K10caeafdff11dad7691d1806694fd82b13c73667ea8fe25e95e2a26a95af0de7c0::server:d3ed21c2fcd1ea570e9209a68039827c INSTALL_K3S_EXEC="server" sh -```

and when i check the service status:
```
● k3s.service - Lightweight Kubernetes
     Loaded: loaded (/etc/systemd/system/k3s.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2024-05-06 21:32:57 IDT; 721ms ago
       Docs: https://k3s.io
    Process: 9924 ExecStartPre=/bin/sh -xc ! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service 2>/dev/>
    Process: 9926 ExecStartPre=/sbin/modprobe br_netfilter (code=exited, status=0/SUCCESS)
    Process: 9927 ExecStartPre=/sbin/modprobe overlay (code=exited, status=0/SUCCESS)
    Process: 9928 ExecStart=/usr/local/bin/k3s server (code=exited, status=1/FAILURE)
   Main PID: 9928 (code=exited, status=1/FAILURE)
        CPU: 449ms
```

i also get the same error when i try to restart the service:
```
Job for k3s.service failed because the control process exited with error code.
See "systemctl status k3s.service" and "journalctl -xeu k3s.service" for details.
```

and when i run:
```
sudo journalctl -u k3s.service --no-pager
```

i get:
```
May 06 21:32:09 pim2 systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
May 06 21:32:09 pim2 systemd[1]: k3s.service: Failed with result 'exit-code'.
May 06 21:32:09 pim2 systemd[1]: Failed to start k3s.service - Lightweight Kubernetes.
May 06 21:32:14 pim2 systemd[1]: k3s.service: Scheduled restart job, restart counter is at 17.
May 06 21:32:14 pim2 systemd[1]: Stopped k3s.service - Lightweight Kubernetes.
May 06 21:32:14 pim2 systemd[1]: Starting k3s.service - Lightweight Kubernetes...
May 06 21:32:14 pim2 sh[9647]: + /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
May 06 21:32:14 pim2 k3s[9651]: time="2024-05-06T21:32:14+03:00" level=info msg="Starting k3s v1.29.4+k3s1 (94e29e2e)"
May 06 21:32:14 pim2 k3s[9651]: time="2024-05-06T21:32:14+03:00" level=info msg="Managed etcd cluster not yet initialized"
May 06 21:32:14 pim2 k3s[9651]: time="2024-05-06T21:32:14+03:00" level=fatal msg="starting kubernetes: preparing server: etcd disabled"
```

any idea what am i doing wrong here ?

this is my current cluster info:
i have a master + worker, and im trying to add another master..
```
NAME   STATUS   ROLES                  AGE    VERSION
pi1    Ready    <none>                 12d    v1.29.3+k3s1
pim1   Ready    control-plane,master   4d3h   v1.29.3+k3s1
```

**Comment by dereknola** [comment]
Your original cluster is utilizing the `sqlite` backend for K3s, which only allows 1 master (better known as `server` today) node. 

If you want to have an HA cluster with multiple K3s servers, you will need to migrate to the embedded etcd backend, then add more server nodes. See https://docs.k3s.io/datastore/ha-embedded#existing-single-node-clusters. 

Note: You must have an odd number of server nodes, you cannot have 2.

**Reply by MosheAzraf** [reply]
the token is the i got from :
```/var/lib/rancher/k3s/server/node-token```
?
